signer/src/storage/postgres.rs

```diff
@@ -946,10 +951,80 @@ impl super::DbRead for PgStore {
 
     async fn get_signer_utxo(
         &self,
-        _chain_tip: &model::BitcoinBlockHash,
-        _aggregate_key: &PublicKey,
+        chain_tip: &model::BitcoinBlockHash,
+        aggregate_key: &PublicKey,
     ) -> Result<Option<SignerUtxo>, Error> {
-        unimplemented!() // TODO(538)
+        let script_pubkey = aggregate_key.signers_script_pubkey();
+
+        let mut txs = sqlx::query_as::<_, model::Transaction>(
+            r#"
+            WITH RECURSIVE tx_block_chain AS (
+                SELECT 
+                    block_hash
+                  , parent_hash
+                FROM sbtc_signer.bitcoin_blocks
+                WHERE block_hash = $1
+
+                UNION ALL
+
+                SELECT
+                    parent.block_hash
+                  , parent.parent_hash
+                FROM sbtc_signer.bitcoin_blocks AS parent
+                JOIN tx_block_chain AS child
+                  ON child.parent_hash = parent.block_hash
+            )
+            SELECT txs.txid, txs.tx, txs.tx_type, tbc.block_hash
+            FROM tx_block_chain AS tbc
+            JOIN sbtc_signer.bitcoin_transactions AS bt ON tbc.block_hash = bt.block_hash
+            JOIN sbtc_signer.transactions AS txs USING (txid)
+            WHERE txs.tx_type = 'sbtc_transaction';
+            "#,
```

Comment:
Should we add a limit on the recursive CTE here? I'm not sure if pg will do something smart (eg, stopping at the first fetch), or first compute the whole temp table and only then looking at the query.

Reply:
Oh yeah, we've typically add a context window to explicitly limit the size of these recursive CTEs in case postgres doesn't do the reasonable thing.